Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp

```diff
@@ -595,6 +605,22 @@ void AcceleratedSurfaceDMABuf::willRenderFrame()
     if (!m_target)
         return;
 
+    if (m_useExplicitSync && m_target->supportsExplicitSync()) {
+        auto fence = m_target->takeReleaseFence();
+        if (fence != -1) {
```

Comment:
We should use UnixFileDescriptor here, either make takeReleaseFence return a UnixFileDescriptor or adopt here the returned fd in a new UnixFileDescriptor

Reply:
Another possibility is doing all the fence wait in the target adding `RenderTarget::waitRelease` or something like that